srunner/scenarios/parking_exit.py

```diff
@@ -0,0 +1,153 @@
+    The ego need to exit from the parking lane and then merge into the driving lane.
+    After the ego is {end_distance} meters away from the parking point, the OutsideRouteLanesTest will be activated and the scenario ends.
+
+    Note 1: For route mode, this shall be the first scenario of the route. The trigger point shall be the first point of the route waypoints.
+
+    Note 2: Make sure there are enough space for spawning blocking vehicles.
+    """
+
+    def __init__(self, world, ego_vehicles, config, debug_mode=False, criteria_enable=True,
+                 timeout=180):
+        """
+        Setup all relevant parameters and create scenario
+        and instantiate scenario manager
+        """
+        self._world = world
+        self._map = CarlaDataProvider.get_map()
+        self.timeout = timeout
+
+        if 'front_vehicle_distance' in config.other_parameters:
+            self._front_vehicle_distance = float(
+                config.other_parameters['front_vehicle_distance']['value'])
+        else:
+            self._front_vehicle_distance = 20  # m
+
+        if 'behind_vehicle_distance' in config.other_parameters:
+            self._behind_vehicle_distance = float(
+                config.other_parameters['behind_vehicle_distance']['value'])
+        else:
+            self._behind_vehicle_distance = 5  # m
+
+        self._end_distance = self._front_vehicle_distance + 5
+
+        if 'parking_lane_side' in config.other_parameters:
+            self._parking_lane_side = config.other_parameters['parking_lane_side']['value']
+        else:
+            self._parking_lane_side = "right"
+
+        super(ParkingExit, self).__init__("ParkingExit",
+                                          ego_vehicles,
+                                          config,
+                                          world,
+                                          debug_mode,
+                                          criteria_enable=criteria_enable)
+
+    def _spawn_blocking_vehicle(self, spawn_point):
+        CarlaDataProvider.request_new_actor(
+            'vehicle.tesla.model3', spawn_point)
+
+    def _create_behavior(self):
+        """
+        Deactivate OutsideRouteLanesTest, then move ego to the parking point, generate blocking vehicles in front of and behind the ego.
+        After ego drives away, actviate OutsideRouteLanesTest, end scenario.
+        """
+
+        sequence = py_trees.composites.Sequence()
+
+        # Put blocking vehicles
+        if self._parking_lane_side == "left":
+            parking_waypoint = self._map.get_waypoint(
+                self.ego_vehicles[0].get_location()).get_left_lane()
+        else:
+            parking_waypoint = self._map.get_waypoint(
+                self.ego_vehicles[0].get_location()).get_right_lane()
+
+        front_points = parking_waypoint.next(self._front_vehicle_distance)
+        if len(front_points) > 0:
+            CarlaDataProvider.request_new_actor(
```

Comment:
It is also common practice to add the to the `self.other_actors` list, to keep track of all spawned vehicles, in case something fails

Reply:
Added